### PR TITLE
Add experiment runner

### PR DESCRIPTION
**Pull request by younesStrittmatter**
added templates and documentation for the experiment runners

### CODE REVIEW

**Review by benwandrew — CHANGES_REQUESTED** [review]
good stuff! clean breakout of the two components constituting the experiment runner. just a few nitpicky suggestions.

File: README.md [hunk 1]
```diff
@@ -48,13 +48,20 @@ conditions $X'$ with respect to a pairwise distance metric applied to existing e
 <br>
 
 (3) **Experiment Runners**: A method that orchestrates the collecting of observations for a given set of 
-experiment conditions.<br>
+experiment conditions (this may include the recruitment of participants).<br>
```
Comment (benwandrew) [inline README.md:51]:
```suggestion
experiment conditions, which may include the recruitment of participants.<br>
```

Comment (benwandrew) [reply README.md:51]: sorry for the naivete, but to contribute an experiment runner is specification of the recruitment manager actually optional in some cases? or does something need to be declared, even if it's the equivalent of None/empty?

File: README.md [hunk 2]
```diff
@@ -48,13 +48,20 @@ conditions $X'$ with respect to a pairwise distance metric applied to existing e
...
 *Example: The [Firebase-Prolific Runner](https://github.com/AutoResearch/autora-experiment-runner-firebase-prolific) 
 enables the collection of behavioral data from human participants via web-based experiments hosted on 
 [Firebase](https://firebase.google.com/), using a pool of participants registered through
 [Prolific](https://www.prolific.co/).* <br>
+(3a) **Recruitment Manager**: A method (or collection of methods) to orchestrate the recruitment of participants.<br>
+*Example: The[Prolific Recruitment Manager](https://github.com/AutoResearch/autora-experiment-runner-recruitment-manager-prolific)
+enables the recruitment of participants via prolific.*<br>
```
Comment (benwandrew) [inline README.md:58]:
```suggestion
enables the recruitment of participants via Prolific.*<br>
```

File: README.md [hunk 3]
```diff
@@ -48,13 +48,20 @@ conditions $X'$ with respect to a pairwise distance metric applied to existing e
 <br>
 
 (3) **Experiment Runners**: A method that orchestrates the collecting of observations for a given set of 
-experiment conditions.<br>
+experiment conditions (this may include the recruitment of participants).<br>
 *Example: The [Firebase-Prolific Runner](https://github.com/AutoResearch/autora-experiment-runner-firebase-prolific) 
 enables the collection of behavioral data from human participants via web-based experiments hosted on 
 [Firebase](https://firebase.google.com/), using a pool of participants registered through
 [Prolific](https://www.prolific.co/).* <br>
+(3a) **Recruitment Manager**: A method (or collection of methods) to orchestrate the recruitment of participants.<br>
```
Comment (benwandrew) [inline README.md:56]:
```suggestion
(3a) **Recruitment Manager**: A method (or collection of methods) to recruit participants.<br>
```

File: README.md [hunk 4]
```diff
@@ -48,13 +48,20 @@ conditions $X'$ with respect to a pairwise distance metric applied to existing e
...
+*Example: The[Prolific Recruitment Manager](https://github.com/AutoResearch/autora-experiment-runner-recruitment-manager-prolific)
+enables the recruitment of participants via prolific.*<br>
+(3b) **Experimentation Manager**: A method (or collection of methods) to orchestrate the experimentation process.<br>
```
Comment (benwandrew) [inline README.md:59]:
```suggestion
(3b) **Experimentation Manager**: A method (or collection of methods) to handle the requisite experimentation processes.<br>
```

File: src/autora/experiment_runner/experimentation_manager/example_experimentation_manager/__init__.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
+"""
+Example experimentation manager:
+    This is an example of functions that could be included
```
Comment (benwandrew) [inline src/autora/experiment_runner/experimentation_manager/example_experimentation_manager/__init__.py:3]:
```suggestion
    These are example functions that could be included 
```

File: src/autora/experiment_runner/experimentation_manager/example_experimentation_manager/__init__.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    This is an example of functions that could be included 
+    (but are not necessarily included)
+"""
+
+
+def send_conditions(args, conditions):
+    """
+    Add a description here
+    Args:
+        args: the arguments needed to set up the experiment (for example credentials)
```
Comment (benwandrew) [inline src/autora/experiment_runner/experimentation_manager/example_experimentation_manager/__init__.py:12]:
```suggestion
        args: the arguments needed to set up the experiment (e.g., credentials)
```

File: src/autora/experiment_runner/recruitment_manager/example_recruitment_manager/__init__.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
+"""
+Example recruitment manager:
+    This is an example of functions that could be included
```
Comment (benwandrew) [inline src/autora/experiment_runner/recruitment_manager/example_recruitment_manager/__init__.py:3]:
```suggestion
    These are example functions that could be included 
```

File: src/autora/experiment_runner/recruitment_manager/example_recruitment_manager/__init__.py [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    This is an example of functions that could be included 
+    (but are not necessarily included)
+"""
+
+
+def example_setup_experiment(args):
+    """
+    Add a description of the setup here
+    Args:
+        the arguments needed to set up the experiment (for example credentials)
```
Comment (benwandrew) [inline src/autora/experiment_runner/recruitment_manager/example_recruitment_manager/__init__.py:12]:
```suggestion
        the arguments needed to set up the experiment (e.g., credentials)
```

**Review by benwandrew — CHANGES_REQUESTED** [review]
good stuff! clean breakout of the two components constituting the experiment runner. just a few nitpicky suggestions.

File: README.md (hunk 1 above)
Comment (musslick) [reply README.md:51]: It's a good point. People may contribute experiment runners without specifying recruitment managers (they would mostly be required for behavioral experiments).

File: src/autora/experiment_runner/experimentation_manager/example_experimentation_manager/__init__.py [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+        conditions: the conditions to be sent
+    """
+    pass
+
+
+def check_status(args):
```
Comment (musslick) [inline src/autora/experiment_runner/experimentation_manager/example_experimentation_manager/__init__.py:18]: Perhaps we could add in the description what the purpose of this function should be?

**Review by musslick — CHANGES_REQUESTED** [review]
Looking good, just have a couple of minor comments. We should also make clear in the README.md that recruitment managers are specific to behavioral experiments. For most, non-behavioral experiments, experimentation_managers should be sufficient.

File: src/autora/experiment_runner/recruitment_manager/example_recruitment_manager/__init__.py [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    """
+    pass
+
+
+def start_experiment(args):
```
[the remark this answers is not in the document]
Comment (younesStrittmatter) [reply src/autora/experiment_runner/recruitment_manager/example_recruitment_manager/__init__.py:17]: Okay makes sense. I'll add example descriptions.

[the remark this answers is not in the document]
Comment (younesStrittmatter) [reply src/autora/experiment_runner/recruitment_manager/example_recruitment_manager/__init__.py]: I changed the function descriptions and names. Maybe this makes things clearer.

File: src/autora/experiment_runner/recruitment_manager/example_recruitment_manager/__init__.py [hunk 11]
```diff
@@ -1,34 +1,48 @@
...
     """
     pass
 
 
-def start_experiment(args):
+def start_recruitment(args):
     """
-    Add description here
+    Add description here.
+    Start the recruitment on an example recrruitment platform.
```
Comment (benwandrew) [inline src/autora/experiment_runner/recruitment_manager/example_recruitment_manager/__init__.py:22]:
```suggestion
    Start the recruitment on an example recruitment platform.
```

File: src/autora/experiment_runner/recruitment_manager/example_recruitment_manager/__init__.py [hunk 12]
```diff
@@ -1,34 +1,48 @@
...
+        the arguments needed to start the recruitment (e.g., credentials)
     """
 
 
-def check_status(args):
+def check_recruitment_status(args):
     """
-    Add description here
+    Add description here.
+    Get information on the status of rectuirment for a study on an example recruitment platform.
```
Comment (benwandrew) [inline src/autora/experiment_runner/recruitment_manager/example_recruitment_manager/__init__.py:43]:
```suggestion
    Get information on the status of recruitment for a study on an example recruitment platform.
```

**Review by benwandrew — CHANGES_REQUESTED** [review]

**Pull request merged**